src/main/scala/karazin/scala/users/group/week1/homework/Homework.scala

```diff
@@ -40,27 +43,85 @@ object Homework :
 
   object `Boolean Operators` :
 
-    def not(b: Boolean): Boolean = ???
+    def not(b: Boolean): Boolean = {
+      if(b == true) false;
+      else
+        true;
+    }
 
-    def and(left: Boolean, right: Boolean): Boolean = ???
-
-    def or(left: Boolean, right: Boolean): Boolean = ???
+    def and(left: Boolean, right: Boolean): Boolean = {
+      if (left == false) false;
+      else if (right == false) false;
+      else true;
+    }
 
+    def or(left: Boolean, right: Boolean): Boolean = {
+      if(left == true) true;
+      else if(right == true) true;
+      else
+        false;
+    }
   end `Boolean Operators`
 
   object `Fermat Numbers` :
 
-    val multiplication: (BigInt, BigInt) => BigInt = ???
+    val multiplication: (BigInt, BigInt) => BigInt = (num1, num2) => {
+      def multiply(num1: BigInt, num2: BigInt, accumulator: BigInt) : BigInt = {
+        if (num1 == 0 || num2 == 0)
+          accumulator
+        else if (num2 < 0)
+          multiply(num1, num2 + 1, accumulator - num1)
+        else
+          multiply(num1, num2 - 1, accumulator + num1)
+      }
+      multiply(num1, num2, 0)
+    }
 
-    val power: (BigInt, BigInt) => BigInt = ???
+    val power: (BigInt, BigInt) => BigInt = (num1, num2) => {
+      def calculatePower(num1: BigInt, num2: BigInt, accumulator: BigInt) : BigInt = {
+        if (num2 > 0)
+          calculatePower(num1, num2 - 1, accumulator * num1)
+        else if (num2 < 0)
+          calculatePower(num1, -num2, 1 / accumulator)
+        else
+          accumulator
+      }
+      calculatePower(num1, num2, 1)
+    }
 
-    val fermatNumber: Int => BigInt = ???
+    val fermatNumber: (Int) => BigInt = (num: Int) => power(BigInt(2), power(BigInt(2), num))+BigInt(1);
 
   end `Fermat Numbers`
 
   object `Look-and-say Sequence` :
-    val lookAndSaySequenceElement: Int => BigInt = ???
+    val lookAndSaySequenceElement: Int => BigInt = (n: Int) =>{
+      if(n <= 0) BigInt(0);
+      var str = "1";
+
+      if (n > 1) str = "11";
 
+      var i = 3;

```

Comment:
Haven't we agreed that `var` is forbidden.
Your solution is not functional. But it should be.